### DISCUSSION TITLE
Calling new script shows blank screen and cursor

### DISCUSSION THREAD

**Discussion by DarkestSoul1992** [ipxe/ipxe#807] [Q&A] [answered]
Using VirtualBox to boot undionly.kpxe from a TFTP server that downloads boot.php and passes username, password and mac address to boot.php before passing control to it. Boot.php authenticates the user, searches for the mac address and sends back (for my test machine) test.ipxe. The problem is that everything works until control is passed to test.ipxe, the screen goes black and I get the text mode cursor but no menu. 
[boot.php.pdf](https://github.com/ipxe/ipxe/files/10035046/boot.php.pdf)
[test.ipxe.pdf](https://github.com/ipxe/ipxe/files/10035049/test.ipxe.pdf)
[embedded.ipxe.pdf](https://github.com/ipxe/ipxe/files/10035061/embedded.ipxe.pdf)

EDIT: Added picture to show the hang and removed both the login screen and framebuffer setup from the embedded script to show easier.
![VirtualBox_PXE Test_18_11_2022_20_56_52](https://user-images.githubusercontent.com/65527568/202807562-2e5cbb04-c10c-490f-b857-697884eb86c2.png)

**Comment by NiKiZe** [comment]
Verify that you are sending a text file starting with "#!ipxe" one not so rare mistake is that you get BOM bytes before that magic marker, often due to UTF-8 encoding.

@mcb30 any chance that we could detect BOM#!ipxe, and if not allow it, maybe show a specific error for it?

**Reply by DarkestSoul1992** [reply]
I have ensured with a hex editor that there are no bytes before #!ipxe

**Reply by stappersg** [reply]
Not knowing what `BOM` is, made me do a websearch.

https://www.ipxe.net/err/2e0080 says that it is a _Byte Order Mark_, typically inserted by a Windows editor.

**Comment by stappersg** [comment]
> The problem is that everything works until control is passed to test.ipxe, the screen goes black and I get the text mode cursor but no menu.

In `test.ipxe.pdf` is
```text
#!ipxe

:retry_setup
console --picture https://wolfnet-computing.com/pxe/background.png || goto retry_setup
```

The _I get text mode cursor_ could be a side-effect of the _retry_setup loop_.

Advice:  Remove that that _loop_.

Request: Report back.

**Reply by DarkestSoul1992** [reply]
It would appear from the logs that it does indeed drop back to text mode just after the script is loaded.
[ipxe.log](https://github.com/ipxe/ipxe/files/10035811/ipxe.log) I also removed the loop and the problem persists.

**Reply by stappersg** [reply]
In that picture is https://ipxe.org/err/3c8c2003 and a few days ago there was https://github.com/ipxe/ipxe/discussions/802 

Advice: Use the advice from #802.

**Reply by DarkestSoul1992** [reply]
Everything was compiled correctly and has displayed picture output using the embedded script. (The login screen has a background picture.) It's when control is passed to test.ipxe that everything crashes and (actively?) drops to text mode.

**Reply by stappersg** [reply]
Acknowledge (and looking forward to further input for this discussion).

**Comment by DarkestSoul1992** [comment, marked as answer]
After adding back in the graphical items to the menus I noticed that I was only loading /computers/test.php, not a fully qualified path. After adding in the fully qualified path, it fired to life on the next test. /computers/test.php IS in the pxe folder on the webserver but is not being loaded from a relative, only an absolute path.
